java/daikon/chicory/Instrument24.java

```diff
@@ -0,0 +1,1720 @@
+  @SuppressWarnings("nullness:initialization")
+  public Instrument24() {
+    super();
+    debug_transform.enabled = Chicory.debug_transform || Chicory.debug || Chicory.verbose;
+    debugInstrument.enabled = Chicory.debug;
+    debug_ppt.enabled = debugInstrument.enabled;
+
+    debug_dir = Chicory.debug_dir;
+    debug_bin_dir = new File(debug_dir, "bin");
+    debug_orig_dir = new File(debug_dir, "orig");
+
+    if (Chicory.dump) {
+      debug_bin_dir.mkdirs();
+      debug_orig_dir.mkdirs();
+    }
+  }
+
+  /**
+   * Returns true if the given ppt should be ignored. Uses the patterns in {@link
+   * daikon.chicory.Runtime#ppt_omit_pattern} and {@link daikon.chicory.Runtime#ppt_select_pattern}.
+   * This method is used by both Chicory and Dyncomp.
+   *
+   * @param className class name to be checked
+   * @param methodName method name to be checked
+   * @param pptName ppt name to be checked
+   * @return true if the item should be filtered out
+   */
+  public static boolean shouldIgnore(String className, String methodName, String pptName) {
+
+    // Don't instrument the class if it matches an excluded regular expression.
+    for (Pattern pattern : Runtime.ppt_omit_pattern) {
+
+      Matcher mPpt = pattern.matcher(pptName);
+      Matcher mClass = pattern.matcher(className);
+      Matcher mMethod = pattern.matcher(methodName);
+
+      if (mPpt.find() || mClass.find() || mMethod.find()) {
+        debug_ppt.log("ignoring %s, it matches ppt_omit regex %s%n", pptName, pattern);
+        return true;
+      }
+    }
+
+    // If any include regular expressions are specified, only instrument
+    // classes that match them
+    if (Runtime.ppt_select_pattern.size() > 0) {
+      for (Pattern pattern : Runtime.ppt_select_pattern) {
+
+        Matcher mPpt = pattern.matcher(pptName);
+        Matcher mClass = pattern.matcher(className);
+        Matcher mMethod = pattern.matcher(methodName);
+
+        if (mPpt.find() || mClass.find() || mMethod.find()) {
+          debug_ppt.log("including %s, it matches ppt_select regex %s%n", pptName, pattern);
+          return false;
+        }
+      }
+    }
+
+    // If we're here, this ppt is not explicitly included or excluded,
+    // so keep unless there were items in the "include only" list.
+    if (Runtime.ppt_select_pattern.size() > 0) {
+      debug_ppt.log("ignoring %s, not included in ppt_select pattern(s)%n", pptName);
+      return true;
+    } else {
+      debug_ppt.log("including %s, not included in ppt_omit pattern(s)%n", pptName);
+      return false;
+    }
+  }
+
+  /**
+   * Given a class, return a transformed version of the class that contains instrumentation code.
+   * Because Chicory is invoked as a javaagent, the transform method is called by the Java runtime
+   * each time a new class is loaded. A return value of null leaves the byte codes unchanged.
+   */
+  @Override
+  public byte @Nullable [] transform(
```

Comment:
Can this be a static method?  How does it depend on instance state of the `Instrument24`?

Reply:
The ClassFileTransformer interface requires transform to be an instance method.  There is only a single instance of Instrument24 for the entire execution of Chicory.